server/channels/app/user.go

```diff
@@ -1704,6 +1704,14 @@ func (a *App) PermanentDeleteUser(c request.CTX, user *model.User) *model.AppErr
 		c.Logger().Warn("You are deleting a user that is a system administrator.  You may need to set another account as the system administrator using the command line tools.", mlog.String("user_email", user.Email))
 	}
 
+	isUserAdminOfTeam, isUserAdminOfTeamErr := a.Srv().Store().Team().IsUserAdminOfTeam(user.Email)
```

Comment:
See my comment on the store method, this should be handled with the permissions. Also, I would try to tackle this issue within the API layer if possible.